### ISSUE TITLE
ministry_of_finance node pool error

### ISSUE THREAD

**Issue by shader** [bcgov/permitify#35]
Progressing gradually through the demo, my next error occurred while using the Ministry of Finance service at localhost:5002:

```
...
ministry_of_finance_1                  |  INFO|indy::commands                |                src/commands/mod.rs:107 | PoolCommand command received
ministry_of_finance_1                  |  INFO|pool_command_executor         |               src/commands/pool.rs:54  | Create command received
ministry_of_finance_1                  | ERROR|indy::errors::indy            |                 src/errors/indy.rs:63  | Casting error to ErrorCode: Pool ledger config already exists Pool ledger config file with 
name "permitify-verifier" already exists                                                                                                                                                                          
ministry_of_finance_1                  | WARNING 2018-01-31 20:15:34,151 libindy 28 140317762131712 _indy_loop_callback: Function returned error 306
ministry_of_finance_1                  | INFO 2018-01-31 20:15:34,152 nodepool 28 140317762131712 Pool config already exists.
ministry_of_finance_1                  |  INFO|indy::commands                |                src/commands/mod.rs:107 | PoolCommand command received
ministry_of_finance_1                  |  INFO|pool_command_executor         |               src/commands/pool.rs:62  | Open command received
ministry_of_finance_1                  | ERROR|indy::errors::indy            |                 src/errors/indy.rs:63  | Casting error to ErrorCode: Invalid Handle: Pool with same name already opened
ministry_of_finance_1                  | WARNING 2018-01-31 20:15:34,154 libindy 28 140317762131712 _indy_loop_callback: Function returned error 301
ministry_of_finance_1                  | WARNING 2018-01-31 20:15:34,154 libindy 28 140317762131712 _indy_loop_callback: Function returned error 301
ministry_of_finance_1                  | ERROR 2018-01-31 20:15:34,154 exception 28 140317762131712 Internal Server Error: /
Traceback:
...
ministry_of_finance_1                  |   File "/app/permitify/views.py", line 40, in index
ministry_of_finance_1                  |     'legal_entity_id': legal_entity_id
ministry_of_finance_1                  |   File "/app/von_connector/proof.py", line 79, in request_proof
ministry_of_finance_1                  |     return eventloop.do(run(filters))
ministry_of_finance_1                  |   File "/app/von_connector/eventloop.py", line 24, in do
ministry_of_finance_1                  |     return event_loop.run_until_complete(coro)
...
ministry_of_finance_1                  |   File "/app/von_connector/proof.py", line 42, in run
ministry_of_finance_1                  |     async with Verifier() as verifier:
ministry_of_finance_1                  |   File "/app/von_connector/agent.py", line 58, in __aenter__
ministry_of_finance_1                  |     await self.pool.open()
ministry_of_finance_1                  |   File "/usr/local/lib/python3.5/dist-packages/von_agent/nodepool.py", line 111, in open
ministry_of_finance_1                  |     self._handle = await pool.open_pool_ledger(self.name, None)
ministry_of_finance_1                  |   File "/usr/local/lib/python3.5/dist-packages/indy/pool.py", line 82, in open_pool_ledger
ministry_of_finance_1                  |     open_pool_ledger.cb)
...
ministry_of_finance_1                  | indy.error.IndyError: ErrorCode.PoolLedgerInvalidPoolHandle
```
(Traceback lines not in permitify code omitted)

It worked when I tried a second time though...

**Comment by nrempel** [comment]
I've seen this now as well.

I believe it occurs when you try to make a connection to the node pool while it is already open.

For several services, a connection is made on page load to verify a proof request. If you try to load the page several times in quick succession you might see this.

Currently, we open and close pool and wallet connections right before and after every interaction with the wallet/ledger.

**Comment by shader** [comment]
Hmm. I ran into that issue on my very first time visiting that service, so I don't _think_ it should been caused by that. Unless it was connecting to the node pool for its own reasons, and I opened the URL too quickly after it started?

Either way, that should probably be insulated a little more from the user. I would have thought that the connection pooling system should be able to handle more than one simultaneous connection though... That or reuse the existing connection to send another message, unless it's using one of ZMQ's req/rep pairs, which cannot send another message until the previous is handled.
